backend/icpy/agent/custom_agent.py

```
@@ -62,85 +52,78 @@ async def _ensure_registry_initialized():
+                        future = executor.submit(init_registry)
+                        agents = future.result(timeout=5)  # 5 second timeout
                         if agents:
                             current_agents = agents
-                except Exception as init_error:
-                    logger.warning(f"Failed to initialize registry: {init_error}")
+                else:
+                    # No running loop, safe to use asyncio.run
+                    agents = asyncio.run(registry.discover_and_load())
+                    if agents:
+                        current_agents = agents
+            except Exception as init_error:
```

Ruff (0.14.3)

Do not catch blind exception: `Exception`

(BLE001)